backend/src/main/java/com/jobwait/spring/SpringEndpoints.java

```diff
@@ -0,0 +1,72 @@
+package com.jobwait.spring;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Supplier;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.CrossOrigin;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.jobwait.control.RequestController;
+import com.jobwait.domain.Answer;
+import com.jobwait.fault.FaultException;
+import com.jobwait.security.AuthToken;
+import com.jobwait.spring.utils.Utils;
+
+// todo - remove all try/catch blocks from jobwait, there should be a single try/catch at the end or whenever we want to convert a runtimeexception/sqlexception into a more generic Fault exception
+@CrossOrigin("http://127.0.0.1:3000")
+@RestController
+public class SpringEndpoints {
+	private static RequestController requestController = new RequestController();
+
+	private static ResponseEntity<ObjectNode> processAndHandleException(Supplier<ObjectNode> func) {
+		ObjectNode responseNode = Utils.mapper.createObjectNode();
+		responseNode.set("data", null);
+		responseNode.set("error", null);
+
+		try {
+			responseNode.set("data", func.get());
+			return new ResponseEntity<ObjectNode>(responseNode, HttpStatus.OK);
+		} catch (FaultException e) {
+			ObjectNode errorNode = Utils.mapper.createObjectNode();
+			errorNode.put("shortCode", e.shortCode);
+			errorNode.put("description", e.description);
+
+			responseNode.set("error", errorNode);
+			return new ResponseEntity<ObjectNode>(responseNode, e.statusCode);
```

Comment:
example of how this looks for an invalid request: 
![image](https://github.com/user-attachments/assets/cb0708fb-ce31-42b9-8ec0-46d9ca0dfcd4)

Reply:
example of how this looks for a valid request:
![image](https://github.com/user-attachments/assets/92c6dbbc-a1ad-4e99-92cb-3afe91dd2e0c)

Reply:
this is how it looks for a request that modifies something (no return data)
![image](https://github.com/user-attachments/assets/d54556ff-3fa8-477a-af62-26351027673f)

Reply:
if you're wondering why "data" and "error" are always present regardless of the status code, I'm thinking it will make frontend work easier to read those keys on every request without worrying if they exist or not